library/core/src/result.rs

```diff
@@ -643,7 +643,8 @@ impl<T, E> Result<T, E> {
     /// ```
     #[inline]
     #[stable(feature = "rust1", since = "1.0.0")]
-    pub fn ok(self) -> Option<T> {
+    #[rustc_const_unstable(feature = "const_result_option", issue = "92384")]
+    pub const fn ok(self) -> Option<T> {
```

Comment:
I'm not too familiar with the feature, but would adding a `~const Drop` bound (either directly to the functions or to a new `impl` block that these functions would then be moved to) work here?
```suggestion
    pub const fn ok(self) -> Option<T>
    where
        T: ~const Drop,
    {
```

Reply:
See also #91928, which `const`ified most of `Option`'s methods.